### PR TITLE
Feature/apollo template

### PR DESCRIPTION
**Pull request by hazzo**
Moved json2typescript models to domain and removed interfaces as they did not add too much to the architecture and created extra files. Typings for api response are generated from graphql schema.

### CODE REVIEW

File: app/templates/README.md.ejs [hunk 1]
```diff
@@ -4,6 +4,14 @@ This repository is a mono repository, this means that there are multiple package
 
 All packages are located inside folder `packages` and every package have a **package.json** file that defines the pkg itself
 
+<%_ if (hasApollo) { -%>
+## Graphql
+First you need to download your schema and update your query types. To do so run the following command:
+```bash
+yarn graphql:build
```
Comment (jfornies) [inline app/templates/README.md.ejs]: maybe this should be `yarn project graphql:build`

File: app/templates/_configs/with-apollo/packages/project/src/accounts/view/components/accounts-list.component.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+  useEffect(() => {
+    void AccountsRepository.getAccountsList().then((response) => {
+      setList(response.data);
+    });
+    void AccountsRepository.getAccountId('1').then((response) => {
+      // eslint-disable-next-line no-console
+      console.log(response)
+    });
+  }, [page]);
```
Comment (jfornies) [inline app/templates/_configs/with-apollo/packages/project/src/accounts/view/components/accounts-list.component.tsx]: `page` is not used in this hook, it isn't a dependency

File: app/templates/packages/project/package.json.ejs [hunk 3]
```diff
@@ -16,6 +16,11 @@
 <%_ } else { -%>
     "start:dev": "run-s translate vite:dev",
     "build": "run-s clean:build build:code",
+<%_ } -%>
+<%_ if (hasApollo) { -%>
+    "graphql:schema": "graphql-codegen --config codegen.yml",
+    "graphql:types": "apollo client:codegen --target=typescript --addTypename",
+    "graphql:build": "yarn graphql:schema && yarn graphql:types",
```
Comment (jfornies) [inline app/templates/packages/project/package.json.ejs]: you can use "npm-run-all"
`"graphql:build": "run-s graphql:schema graphql:types",`

File: app/templates/packages/project/package.json.ejs [hunk 4]
```diff
@@ -45,6 +50,10 @@
     "@sentry/react": "6.2.3",
 <%_ if (hasAxios) { -%>
     "axios": "0.21.1",
+<%_ } -%>
+<%_ if (hasApollo) { -%>
+    "@apollo/client": "^3.3.18",
```
Comment (jfornies) [inline app/templates/packages/project/package.json.ejs]: remove `^` modifier

File: app/templates/packages/project/package.json.ejs [hunk 5]
```diff
@@ -72,6 +81,11 @@
     "@types/react-dom": "17.0.3",
     "@types/react-router-dom": "5.1.7",
     "@vitejs/plugin-react-refresh": "1.3.1",
+<%_ if (hasApollo) { -%>
+    "@graphql-codegen/cli": "1.21.4",
+    "@graphql-codegen/schema-ast": "^1.18.2",
```
Comment (jfornies) [inline app/templates/packages/project/package.json.ejs]: remove `^` modifier

File: app/templates/packages/project/package.json.ejs (hunk 3 above)
Comment (hazzo) [reply app/templates/packages/project/package.json.ejs]: && makes commands sequentially but if `run-s` is used along the project I can change it.

File: app/templates/_configs/with-apollo/packages/project/src/accounts/view/components/accounts-list.component.tsx (hunk 2 above)
Comment (hazzo) [reply app/templates/_configs/with-apollo/packages/project/src/accounts/view/components/accounts-list.component.tsx]: Yes this is from old axios implementation as the graphql api does not have pages.

File: app/templates/README.md.ejs (hunk 1 above)
Comment (hazzo) [reply app/templates/README.md.ejs]: Yeah to avoid user navigating to project folder. Good point.

**Pull request merged**